### PR TITLE
Parse email addresses with non-ASCII UTF8 domain parts

### PR DESCRIPTION
**Pull request by rsto**
Email addresses having non-ASCII UTF8 domain parts up until now were parsed incorrectly, leading to truncated email addresses being stored in the mailbox cache and returned in JMAP and IMAP.

This fixes the email address parser to parse such domains and bumps the mailbox minor cache version number.

### CODE REVIEW

**Comment by rsto** [comment]
Huh, that success locally but fails in CI. Changing this to Draft until I fixed that.

File: lib/parseaddr.c [hunk 1]
```diff
@@ -314,7 +314,7 @@ static int parseaddr_domain(char **inp, char **domainp, char **commentp, int *in
 
     for (;;) {
         c = *src++;
-        if (Uisalnum(c) || c == '-' || c == '[' || c == ']' || c == ':') {
+        if (Uisalnum(c) || c == '-' || c == '[' || c == ']' || c == ':' || c > 127) {
             *dst++ = c;
```
Comment (ksmurchison) [inline lib/parseaddr.c:318]: Do we care to make sure that the char is in the (0x80 <= c <= 0xF4)  range?

Comment (rsto) [reply lib/parseaddr.c:318]: I'd say not to. Unless we change the parseaddr to check if the byte sequence is truly valid UTF-8, we should leave that to the client.

**Comment by rsto** [comment]
Thanks for the `int` vs `u_char` fix. Interesting that this didn't fail on my setup.

**Review by ksmurchison — APPROVED** [review]

**Pull request merged**